packages/core/src/command/consolidated/consolidated-filter-by-date.ts

```diff
@@ -0,0 +1,334 @@
+      case "MedicationDispense":
+        return isMedicationDispenseWithinDateRange(entry.resource, range);
+      case "MedicationRequest":
+        return isMedicationRequestWithinDateRange(entry.resource, range);
+      case "MedicationStatement":
+        return isMedicationStatementWithinDateRange(entry.resource, range);
+      case "Observation":
+        return isObservationWithinDateRange(entry.resource, range);
+      case "Procedure":
+        return isProcedureWithinDateRange(entry.resource, range);
+      case "Provenance":
+        return isProvenanceWithinDateRange(entry.resource, range);
+      case "RelatedPerson":
+        return isRelatedPersonWithinDateRange(entry.resource, range);
+    }
+    return includeIfNoDate;
+  });
+  return {
+    ...bundle,
+    total: filteredEntries?.length ?? 0,
+    entry: filteredEntries ?? [],
+  };
+}
+
+function isAllergyIntoleranceWithinDateRange(
+  resource: AllergyIntolerance,
+  range: DateRange
+): boolean {
+  if (resource.onsetPeriod) {
+    const res = arePeriodsWithinRange([resource.onsetPeriod], range);
+    if (res !== undefined) return res;
+  }
+  if (resource.onsetRange) {
+    const res = areRangesWithinRange([resource.onsetRange], range);
+    if (res !== undefined) return res;
+  }
+  if (resource.onsetString) {
+    const res = isDateWithinDateRange(resource.onsetString, range);
+    if (res !== undefined) return res;
+  }
+  if (resource.onsetDateTime) {
+    const res = isDateWithinDateRange(resource.onsetDateTime, range);
+    if (res !== undefined) return res;
+  }
+  if (resource.lastOccurrence) {
+    const res = isDateWithinDateRange(resource.lastOccurrence, range);
+    if (res !== undefined) return res;
+  }
+  return includeIfNoDate;
+}
+
+function isCommunicationWithinDateRange(resource: Communication, range: DateRange): boolean {
+  const res = areDatesWithinRange([resource.sent, resource.received], range);
+  if (res !== undefined) return res;
+  return includeIfNoDate;
+}
+
+function isCompositionWithinDateRange(resource: Composition, range: DateRange): boolean {
+  const periods = resource.event?.map(e => e.period);
+  const res = arePeriodsWithinRange(periods, range);
+  if (res !== undefined) return res;
+  return includeIfNoDate;
+}
+
+function isConditionWithinDateRange(resource: Condition, range: DateRange): boolean {
+  if (resource.onsetPeriod) {
+    const res = arePeriodsWithinRange([resource.onsetPeriod], range);
+    if (res !== undefined) return res;
+  }
+  if (resource.onsetRange) {
+    const res = areRangesWithinRange([resource.onsetRange], range);
+    if (res !== undefined) return res;
+  }
+  if (resource.onsetString) {
+    const res = isDateWithinDateRange(resource.onsetString, range);
+    if (res !== undefined) return res;
+  }
+  if (resource.onsetDateTime) {
+    const res = isDateWithinDateRange(resource.onsetDateTime, range);
+    if (res !== undefined) return res;
```

Comment:
missing onsetAge

Reply:
Age following [this comment](https://github.com/metriport/metriport/pull/2756#discussion_r1764259029) - added TODO just in case.